codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -145,6 +147,17 @@ public ChannelFuture writeHeaders(ChannelHandlerContext ctx, int streamId, Http2
         return writeHeaders(ctx, streamId, headers, 0, DEFAULT_PRIORITY_WEIGHT, false, padding, endStream, promise);
     }
 
+    private static boolean validateHeadersSentState(Http2Stream stream, Http2Headers headers, boolean isServer,
+                                                    boolean endOfStream) {
+        boolean isInformational = false;
+        if ((isServer && (isInformational = HttpStatusClass.valueOf(headers.status()) == INFORMATIONAL)
```

Comment:
+1. It can be looks like this:
```java
boolean isInformational = isServer && HttpStatusClass.valueOf(headers.status()) == INFORMATIONAL;
if ((isInformational || !endOfStream) && stream.isHeadersSent() || stream.isTrailersSent()) {
    ...
}
```